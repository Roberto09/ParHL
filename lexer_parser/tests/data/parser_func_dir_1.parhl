let x : int := 10, y[10] : float, z : string := "abc" 

let my_func (x : int, k : float) : int {
    print(x + y[1][2][3])
    let my_super_func(x : float, z: string) : string{
        print(123);
    }
    while(true){
        let a : int := 10
        if (x > 10){
            print(124)
        }
    }
}


