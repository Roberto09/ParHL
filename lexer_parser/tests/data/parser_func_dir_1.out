FuncDir - func_stack: [(name: glob, type: void, vars: [(name: x, type: int), (name: y, type: float), (name: z, type: string)], funcs: [(name: my_func, type: int, vars: [(name: x, type: int), (name: k, type: float)], funcs: [(name: my_super_func, type: string, vars: [(name: x, type: float), (name: z, type: string)], funcs: [], blocks:[])], blocks:[(block, vars: [(name: a, type: int)], funcs: [], blocks:[(block, vars: [], funcs: [], blocks:[]))]))])], blocks:[])]
